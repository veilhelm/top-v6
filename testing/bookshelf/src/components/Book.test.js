import React from 'react'
import { render, cleanup } from '@testing-library/react'
import { Book } from './Book.js'

describe('Book', () => {
  afterEach(cleanup)

  it('should render a book correctly', () => {
    const book = {
      title: 'a title',
      description: 'a very long description lalalalalala',
      votes: 3,
    };

    const { getByText } = render(
      <Book 
        title={book.title}
        description={book.description}
        votes={book.votes}
      />
    )

    expect(getByText(book.description)).toBeInTheDocument();
    expect(getByText(book.title)).toBeInTheDocument();
    expect(getByText(`votes: ${book.votes}`)).toBeInTheDocument();
  })

  it('should match snapshot', () => {
    const book = {
      title: 'a title',
      description: 'a very long description lalalalalala',
      votes: 3,
    };

    const { container } = render(
      <Book 
        title={book.title}
        description={book.description}
        votes={book.votes}
      />
    )

    expect(container).toMatchSnapshot()
  })
})
