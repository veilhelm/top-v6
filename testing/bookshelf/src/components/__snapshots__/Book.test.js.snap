// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book should match snapshot 1`] = `
<div>
  <div
    class="book"
    data-testid="book"
  >
    <h2
      class="book-title"
    >
      a title
    </h2>
    <span>
      votes: 
      3
    </span>
    <p>
      a very long description lalalalalala
    </p>
  </div>
</div>
`;
